circuits/bid/src/correctness.rs

```diff
@@ -123,28 +133,65 @@ mod tests {
             blinder: blinder,
         };
 
-        // Generate Composer & Public Parameters
-        let pub_params = unsafe {
-            PublicParameters::from_slice_unchecked(
-                rusk_profile::get_common_reference_string()
-                    .expect("Failed to fetch CRS from rusk_profile")
-                    .as_slice(),
-            )
-        };
-
-        let (pk, vd) = circuit.compile(&pub_params)?;
-        let proof = circuit.gen_proof(&pub_params, &pk, b"BidCorrectness")?;
+        let proof = circuit.gen_proof(&PP, &KEYS.0, b"BidCorrectness")?;
 
         let pi = vec![commitment.into()];
         assert!(circuit::verify_proof(
-            &pub_params,
-            &vd.key(),
+            &PP,
+            &KEYS.1.key(),
             &proof,
             &pi,
-            &vd.pi_pos(),
+            &KEYS.1.pi_pos(),
             b"BidCorrectness",
         )
         .is_err());
         Ok(())
     }
+
+    #[test]
+    fn test_correctness_with_message() -> Result<(), Error> {
+        use dusk_blindbid::Bid;
+        use dusk_pki::{PublicSpendKey, SecretSpendKey};
+        use dusk_poseidon::sponge;
+        use rand::Rng;
+
+        let mut rng = rand::thread_rng();
+        let secret = JubJubScalar::random(&mut rand::thread_rng());
+        let secret_k = BlsScalar::random(&mut rand::thread_rng());
+        let psk = PublicSpendKey::from(SecretSpendKey::random(
+            &mut rand::thread_rng(),
+        ));
+        let value: u64 =
+            (&mut rand::thread_rng()).gen_range(V_RAW_MIN..V_RAW_MAX);
+
+        let r = JubJubScalar::random(&mut rand::thread_rng());
+        let bid = Bid::new(
+            Message::new(&mut rng, &secret, &psk, value),
+            sponge::sponge::sponge_hash(&[secret_k]),
+            psk.gen_stealth_address(&secret),
+            u64::MAX,
+            u64::MAX,
+        );
+
+        let (value, blinder) =
+            bid.decrypt_data(&r, &psk).expect("decryption error");
```

Comment:
This wouldn't work. You create the message with `secret` and decrypt with `r`

Reply:
Was an error. I missed the replacement